web_console/components/Form.jsx

```diff
@@ -40,28 +40,60 @@ export default function Form({
   // cache raw fields data
   const rawFields = fields
   // TODO: handle name confilicts
+
   const groupFormType = useMemo(() => ({}), [])
-  // flat all group fileds
-  fields = fields.reduce((total, curr) => {
-    if (curr.groupName) {
-      if (Array.isArray(curr.fields)) {
-        total.push(...curr.fields)
-      } else {
-        Object.values(curr.fields).forEach(el => total.push(...el))
-      }
-    } else {
-      total.push(curr)
-    }
-    return total
-  }, [])
   const theme = useTheme();
   const styles = useStyles(theme);
-  const [form, setForm] = useState(fields.reduce((total, current) => {
-    total[current.key] = current.hasOwnProperty('value')
-      ? current.value
-      : current.value || current.default;
-    return total;
-  }, {}));
+
+  const mapFields2Form = fields => {
+    // flat all group fileds
+    fields = fields.reduce((total, curr) => {
```

Comment:
why update `fields` directly?

Reply:
cause `fields` are used in existing code, update it to feet them. It shouldn't be implemented like this, i'll update later